plugin/db/mysql/mysql.go

```diff
@@ -940,6 +944,114 @@ func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner) (err error
 	return nil
 }
 
+// RestoreIncremental restores the database using incremental backup in time range of [config.Start, config.End).
+func (driver *Driver) RestoreIncremental(ctx context.Context, config db.RecoveryConfig) error {
+	return fmt.Errorf("Unimplemented")
+}
+
+// RestorePITR is a wrapper for restore a full backup and a range of incremental backup
+func (driver *Driver) RestorePITR(ctx context.Context, fullBackup *bufio.Scanner, config db.RecoveryConfig, database string, timestamp int64) error {
+	pitrDatabaseName := getPITRDatabaseName(database, timestamp)
+	if _, err := driver.db.Exec(fmt.Sprintf("CREATE DATABASE %s", pitrDatabaseName)); err != nil {
+		return err
+	}
+
+	if _, err := driver.db.Exec(fmt.Sprintf("USE %s", pitrDatabaseName)); err != nil {
+		return err
+	}
+
+	if _, err := driver.db.Exec("SET foreign_key_checks=OFF"); err != nil {
+		return err
+	}
+
+	if err := driver.Restore(ctx, fullBackup); err != nil {
+		return err
+	}
+
+	// TODO(dragonly): implement RestoreIncremental in mysql driver
+	_ = driver.RestoreIncremental(ctx, config)
+
+	if _, err := driver.db.Exec("SET foreign_key_checks=ON"); err != nil {
+		return err
+	}
+
+	if _, err := driver.db.Exec(fmt.Sprintf("USE %s", database)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// SwapPITRDatabase renames the pitr database to the target, and the original to the old database
+func (driver *Driver) SwapPITRDatabase(ctx context.Context, database string, timestamp int64) error {
+	txn, err := driver.db.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer txn.Rollback()
+
+	pitrOldDatabase := getPITRDatabaseOldName(database)
+	pitrDatabaseName := getPITRDatabaseName(database, timestamp)
+
+	if _, err := txn.ExecContext(ctx, fmt.Sprintf("CREATE DATABASE %s", pitrOldDatabase)); err != nil {
+		return err
+	}
+
+	tables, err := getTables(txn, database)
+	if err != nil {
+		return fmt.Errorf("failed to get tables of database %q, error[%w]", database, err)
+	}
+	tablesPITR, err := getTables(txn, pitrDatabaseName)
+	if err != nil {
+		return fmt.Errorf("failed to get tables of database %q, error[%w]", pitrDatabaseName, err)
+	}
+
+	var buf strings.Builder
+	buf.WriteString("RENAME TABLE")
+	for _, table := range tables {
+		buf.WriteString(fmt.Sprintf(" `%s`.`%s` TO `%s`.`%s`,", database, table.name, pitrOldDatabase, table.name))
```

Comment:
It might be easier to do

```
var tableRenames []string
for ... {
  tableRenames = append(tableRenames, xxx)
}
renameStatement := fmt.Sprintf("RENAME TABLE %s;", strings.Join(tableRenames, " "))
```

Let's not worry about performance. It's not easy to get string buffer writes right without unit tests. Readability and simplicity is more important than tiny performance improvement.